bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/type/HomematicTypeGeneratorImpl.java

```diff
@@ -279,8 +279,9 @@ public StateOption createOption(String value, String description) {
                 BigDecimal min = MetadataUtils.createBigDecimal(dp.getMinValue());
                 BigDecimal max = MetadataUtils.createBigDecimal(dp.getMaxValue());
                 BigDecimal step = MetadataUtils.createBigDecimal(dp.getStep());
-                if (ITEM_TYPE_DIMMER.equals(itemType) && dp.getMaxValue().doubleValue() == 1.01d) {
-                    // For dimmers with a max value of 1.01 the values must be corrected
+                if (ITEM_TYPE_DIMMER.equals(itemType)
+                        && (dp.getMaxValue().doubleValue() == 1.01d || dp.getMaxValue().doubleValue() == 1.0d)) {
+                    // For dimmers with a max value of 1.01 or 1.0 the values must be corrected
```

Comment:
`float` and `double` shouldn't be compare with `==`. You could use `>=` and `<=`.

Reply:
I agree with you, that == is normally not a good idea for float/double. In this case I tested it with these values, but I changed the comparison to make it more safe.

Reply:
Wouldn't it be better to create a `new BigDecimal(dp.getMaxValue().toString())` and use `compareTo(new BigDecimal("1.01"))`? Then the compared values would still make sense for the reader of the code.

Reply:
I thought about using BigDecimal for the comparison but I wanted to avoid the overhead.

Reply:
I've checked the code and maxValue from `HmDataPoint()` is already converted to `BigDecimal()` at that point. Therefore we could simply use the `max` variable for comparism.
`max.compareTo(BigDecimal.valueof("1.0") == 0 || max.compareTo(BigDecimal.valueof("1.01") == 0`

Maybe that is a better solution. What do you think?

Reply:
Hmm, I think this would be similar to my original solution with "==".

Reply:
`max` is `BigDecimal()` and therefore [compareTo()](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/math/BigDecimal.html#compareTo(java.math.BigDecimal)) would honor it's value properly instead of "==".

Reply:
How should we proceed? I would really like to see this fix in OH3 before it will be released.

Reply:
I agree to @xes0r this is even more elegeant. The final version of OH3 is probably already built. In any case there's a complete freeze. So, this will probably go into 3.1.

Reply:
OK, you conviced me. I have changed it to compareTo (valueOf is not available for String, instead a constructor must be used)

Reply:
@fwolter I have changed the coding according to your proposal. Maybe you can find some time to review it again (and hopefully merge it).